packages/cli/src/util/update-notifier/index.ts

```diff
@@ -57,33 +60,47 @@ export default async function updateNotifier({
       // signal the worker to fetch the latest version
 
       // we need to find the update worker script since the location is
-      // different based on production vs unit tests
+      // different based on production vs tests because
       let dir = dirname(__filename);
       let script = resolvePath(dir, 'dist', 'update-worker.js');
       const { root } = parsePath(dir);
       while (!(await pathExists(script))) {
         dir = dirname(dir);
         if (dir === root) {
           // didn't find it, bail
+          output?.debug('Failed to find update worker script!');
           resolve();
           return;
         }
         script = resolvePath(dir, 'dist', 'update-worker.js');
       }
 
+      // spawn the worker with an IPC channel
       output?.debug(`Spawning ${script}`);
       const worker = spawn(process.execPath, [script], {
         stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
         windowsHide: true,
       });
 
+      // we allow the child 2 seconds to let us know it's ready before we give up
+      const timer = setTimeout(() => {
```

Comment:
**suggestion (non-blocking):** rename `timer` -> `workerReadyTimer`

Can we use a more specific name for the timeout handle? It will make it clearer what the `clearTimeout` line is doing, for example:

```
clearTimeout(workerReadyTimer);
```